third_party/terraform/tests/resource_compute_network_endpoint_test.go.erb

```diff
@@ -0,0 +1,243 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+import (
+	"testing"
+
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"log"
+)
+
+// TODO(emilyye): Fix this test once network endpoint and group are GA.
+// Terraform will add an implicit provider dependency on the GA provider
+// instead of the non-GA provider if the resource no longer exists in the
+// test step config (and no longer has the explicit provider declaration).
+func TestAccComputeNetworkEndpoint_networkEndpointsBasic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(10),
+	}
+
+	initialEndpt := &networkEndptTestId{}
+	modifiedEndpt := &networkEndptTestId{}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProvidersOiCS,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeNetworkEndpoint_networkEndpointsBasic(context),
```

Comment:
Can you add a case before this with multiple NEGs, so we test the mutex & test deleting a single NEG when we move to this step?

Reply:
nope, not until we get to GA provider tests! If you delete any of them, the next step automatically assumes the resource that was removed from config needs provider "google". 

I can add multiple to test the mutex, but honestly I was just gonna add more tests that are actually good once I switch this over to GA provider.

Reply:
(both NEG and endpoints are in GA API already)

Reply:
Since we probably want to add a multi-stage test for this regardless of version, should we use the normal acceptance test provider set here so that `google` is whichever version of the provider we're using?

Reply:
oh, i misunderstood what testAccProviders/OiCS does - so if I just remove the google-beta blocks it will work and still use the google-beta provider in the test?

Reply:
Yep! Using the OiCS provider set tests that the `google-beta` provider is specified on the resources explicitly, to make sure the OiCS examples work.

Reply:
:+1: fixed test then